flow.c

```diff
@@ -16,17 +17,19 @@ static struct fio_sem *flow_lock;
 int flow_threshold_exceeded(struct thread_data *td)
 {
 	struct fio_flow *flow = td->flow;
-	long long flow_counter;
+	double flow_counter_ratio, flow_weight_ratio;
 
 	if (!flow)
 		return 0;
 
-	if (td->o.flow > 0)
-		flow_counter = flow->flow_counter;
-	else
-		flow_counter = -flow->flow_counter;
+	flow_counter_ratio = (double)td->flow_counter / atomic_load_relaxed(&flow->flow_counter);
```

Comment:
not sure if should just replace atomic loads with memory barrier

Reply:
I prefer atomic_load_relaxed() since using that primitive makes it clear which variable is shared with other threads. That is not the case when using memory barriers.